### PR TITLE
RUMM-206 Add the benchmarks for Tracing API

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

This PR adds some benchmarks scenarios related with the newly added **Trace** Api.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: instrumented/benchmark/build.gradle.kts [hunk 1]
```diff
@@ -100,12 +100,55 @@ kotlinConfig()
...
+
+    addThreshold(
+        "benchmark_sending_medium_load_of_logs",
+        TimeUnit.MILLISECONDS.toNanos(200)
+    )
+    addThreshold(
+        "benchmark_sending_heavy_load_of_logs",
+        TimeUnit.MILLISECONDS.toNanos(300)
+    )
+}
```
Comment (xgouchet) [inline instrumented/benchmark/build.gradle.kts]: Why do you create a second `reviewBenchmark` block ? That's a bit unusual, why not have a single block and just add comment to distinguish logs from traces ?

File: instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/trace/TraceApiBenchmark.kt [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    @After
+    fun tearDown() {
+        Datadog.invokeMethod("stop")
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun benchmark_sending_span() {
```
Comment (xgouchet) [inline instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/trace/TraceApiBenchmark.kt]: it's not `sending` anything, only `creating` or `writing` a span :)

File: instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/trace/TraceApiBenchmark.kt [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    fun benchmark_sending_medium_load_of_spans() {
+        addSpans(MEDIUM_ITERATIONS)
+    }
+
+    @Test
+    fun benchmark_sending_heavy_load_of_spans() {
+        addSpans(BIG_ITERATIONS)
+    }
+
+    private fun addSpans(iterations: Int) {
```
Comment (xgouchet) [inline instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/trace/TraceApiBenchmark.kt]: this doesn't add anything, you could rename it `benchmarkCreatingLogsWithTagsAndMatadata`

File: instrumented/benchmark/build.gradle.kts (hunk 1 above)
Comment (mariusc83) [reply instrumented/benchmark/build.gradle.kts]: was hesitating, I thought is much more visible this way....will put it back, you're right ;)

**Pull request merged**